R/setup.R

```diff
@@ -92,148 +114,196 @@ setup <- function(x_train,
 
   internal <- check_and_set_parameters(internal)
 
+  internal <- set_adaptive_parameters(internal,prev_iter_list)
+
+  internal$timing_list <- list(
+    init_time = init_time,
+    setup = Sys.time()
+  )
+
   return(internal)
 }
 
 #' @keywords internal
-check_and_set_parameters <- function(internal) {
-  # Check groups
-  feature_names <- internal$parameters$feature_names
-  group <- internal$parameters$group
-  n_combinations <- internal$parameters$n_combinations
-  n_features <- internal$parameters$n_features
-  n_groups <- internal$parameters$n_groups
-  is_groupwise <- internal$parameters$is_groupwise
-  exact <- internal$parameters$exact
-
-  if (!is.null(group)) check_groups(feature_names, group)
+get_parameters <- function(approach, paired_shap_sampling, prediction_zero, output_size = 1, max_n_coalitions, group,
+                           n_MC_samples, seed, keep_samp_for_vS, type, horizon, train_idx, explain_idx,
+                           explain_y_lags, explain_xreg_lags, group_lags = NULL, MSEv_uniform_comb_weights,
+                           verbose, adaptive = FALSE, adaptive_arguments = adaptive_arguments,
+                           shapley_reweighting = "none", testing, is_python, ...) {
```

Comment:
I would include all possibilities for `shapley_reweighting` and then test with `match.arg`

Reply:
As above, I agree it is better, but I'll keep it for now anyhow. If you/we want to, we can go over things like this in a separate PR later.